capstone/src/main/webapp/script.js

```diff
@@ -49,3 +49,137 @@ function  formatCardInHTML(studySet) {
       </div>
     </a>`;
 }
+
+function addCard() {
+  var form = document.getElementById("card_form");
+
+  var newFrontCard = document.createElement("div");
+  newFrontCard.setAttribute("class", "input-field col s6");
+
+  var frontCard = document.createElement("input");
+  frontCard.setAttribute("id", "front");
+  frontCard.setAttribute("type", "text");
+  frontCard.setAttribute("class", "validate input-card");
+
+  var frontLabel = document.createElement("label");
+  frontLabel.setAttribute("for", "front");
+  var frontText = document.createTextNode("Front");
+  frontLabel.append(frontText);
+
+  newFrontCard.append(frontCard);
+  newFrontCard.append(frontLabel);
+
+  form.appendChild(newFrontCard);
+
+  var newBackCard = document.createElement("div");
+  newBackCard.setAttribute("class", "input-field col s6");
+
+  var backCard = document.createElement("input");
+  backCard.setAttribute("id", "back");
+  backCard.setAttribute("type", "text");
+  backCard.setAttribute("class", "validate input-card");
+
+  var backLabel = document.createElement("label");
+  backLabel.setAttribute("for", "back");
+  var backText = document.createTextNode("Back");
+  backLabel.append(backText);
+
+  newBackCard.append(backCard);
+  newBackCard.append(backLabel);
+
+  form.appendChild(newBackCard);
+}
+
+function addStudySet() {
+  var elements = document.getElementsByClassName("input-card");
+  var cards = [];
+
+  for (var i = 0; i < elements.length - 1; i = i + 2) {
+    var contacts = { front: elements[i].value, back: elements[i + 1].value };
+    cards.push(contacts);
+  }
+
+  var user_id = document.getElementById("user_id").value;
+  user_id = user_id.trim();
+
+  var title = document.getElementById("title").value;
+  title = title.trim();
+
+  var subject = document.getElementById("subject").value;
+  subject = subject.trim();
+
+  var description = document.getElementById("description").value;
+  description = description.trim();
+
+  var university_id = document.getElementById("university_id").value;
+  university_id = university_id.trim();
+
+  var professor = document.getElementById("professor").value;
+  professor = professor.trim();
+
+  var academic_time = document.getElementById("academic_time").value;
+  academic_time = academic_time.trim();
+
+  var course_name = document.getElementById("course_name").value;
+  course_name = course_name.trim();
+
+  if (
+    !isStudySetInfoFilled(
+      user_id,
+      title,
+      subject,
+      description,
+      university_id,
+      professor,
+      academic_time,
+      course_name
+    )
+  ) {
+    window.alert(
+      "Please make sure that all study set information has been filled in"
+    );
+    return;
+  }
```

Comment:
My opinion, to avoid all this and the isStudySetInfoFIlled() method below, just add the "required" keyword to your html form tags and that will take care of these lines. 

Also considering some browsers or users disable pop-up menus, the window.alert() trigger on line 139 might not be seen.

Reply:
I'm pretty sure the required tag does not work because the inputs are not under a tag.